src/graph.rs

```diff
@@ -1080,6 +1082,46 @@ impl PyGraph {
         neighbors.count()
     }
 
+    /// Generate a new :class:`~retworkx.PyDiGraph` object from this graph
+    ///
+    /// This will create a new :class:`~retworkx.PyDiGraph` object from this
+    /// graph. All edges in this graph will result in a bidirectional edge
+    /// pair in the output graph.
+    ///
+    /// :returns: A new :class:`~retworkx.PyDiGraph` object with a
+    ///     bidirectional edge pair for each edge in this graph. Also all
+    ///     node and edge weights/data payloads are copied by reference to
+    ///     the output graph
+    /// :rtype: PyDiGraph
+    pub fn to_directed(&self, py: Python) -> crate::digraph::PyDiGraph {
+        let node_count = self.node_count();
+        let mut new_graph = StableDiGraph::<PyObject, PyObject>::with_capacity(
+            node_count,
+            2 * self.graph.edge_count(),
+        );
+        let mut node_map: HashMap<NodeIndex, NodeIndex> =
+            HashMap::with_capacity(node_count);
+        for node_index in self.graph.node_indices() {
+            let node = self.graph[node_index].clone_ref(py);
+            let new_index = new_graph.add_node(node);
+            node_map.insert(node_index, new_index);
```

Comment:
Node indices might change and I think it is worth documenting it

Reply:
Good call, I added a note to the docstring in: https://github.com/Qiskit/retworkx/pull/419/commits/706e07decdbb76abe8129ae9f038341b11f3f2ef (also for `to_undirected()`, which had the same thing)